src/django_components/slots.py

```diff
@@ -70,17 +77,30 @@ class Slot(Generic[TSlotData]):
     slot_name: Optional[str] = None
     """Name of the slot that originally defined or accepted this slot fill."""
     nodelist: Optional[NodeList] = None
-    """Nodelist of the slot content."""
+    """If the slot was defined with `{% fill %} tag, this will be the Nodelist of the slot content."""
 
     def __post_init__(self) -> None:
+        # Since the `Slot` instance is treated as a function, it may be passed as `contents`
```

Comment:
This is the main part of the PR 

- Before, the slot input was normalized when calling `Component.render()`, inside [`_normalize_slot_fills()`](https://github.com/django-components/django-components/pull/1180/files#diff-35c84ebceae50f7149f03bbcaf2b4f1289bc2984bf7859ca521b5dbe7960c396R2707)

- Now the main normalization logic is `Slot()` constructor, so people will be able to call `Slot(...)` from wherever and still get the same behaviour.